lib/xls_export.rb

```diff
@@ -393,12 +398,17 @@ def issues_to_xls2(issues, project, query, options = {})
         return xls_stream.string
       end
 
-      def journal_details_to_xls(issue, options)
+      def journal_details_to_xls(issue, options, book = nil)
         issue_updates = issue.journals.includes(:user, :details).order("#{Journal.table_name}.created_on ASC").to_a
         return nil if issue_updates.size == 0
 
         Spreadsheet.client_encoding = 'UTF-8'
-        book = Spreadsheet::Workbook.new
+        
+        write = false

```

Comment:
I think a condition that **options[:journal_worksheets] is false** is better than the flag **write** .
It is more clear that journal books are created when the option is unchecked.

Reply:
I agree that your idea is clearer for the reader. However, I have a concern with your suggestion detailed below.

If the caller doesn't provide any book (i.e. book=nil) but **options[:journal_worksheets]** is **false** then there will be an exception at the line **book.write(xls_stream)**.

Therefore I think the code I wrote is a little more rhobust: if no book is provided then we always create one irrespective of the options. Please let me know your comments or suggestions.

Reply:
Thank you, your code is very clear and it removes my concern. I think that your ruby is a lot better than mine too!